src/v2/Utils/Hooks/useTimer.tsx

```diff
@@ -25,16 +25,19 @@ const extractTime = (time: number) => {
 export const useTimer = (endDate: string, startAt: string = ""): Timer => {
   const currentTime = useCurrentTime({ syncWithServer: true })
 
-  const duration = Duration.fromISO(
+  const timeBeforeEnd = Duration.fromISO(
     DateTime.fromISO(endDate).diff(DateTime.fromISO(currentTime)).toString()
   )
-  const hasEnded = Math.floor(duration.seconds) <= 0
+  const hasEnded = Math.floor(timeBeforeEnd.seconds) <= 0
 
   const timeBeforeStart = Duration.fromISO(
     DateTime.fromISO(startAt).diff(DateTime.fromISO(currentTime)).toString()
   )
   const hasStarted = Math.floor(timeBeforeStart.seconds) <= 0
 
+  const duration =
```

Comment:
Ok, so if this hook knows that the timed event hasn't started yet, it returns that duration, otherwise it returns the duration until it has ended. That does meaningfully change _how_ this hook is used, but I see that it is somewhat opt-in (since you need to have provided the optional `startAt` in the first place). I think that's fine.

Reply:
I pretty thoroughly checked around to see where this was being used and I think we should be good.